File: HW4/Cube.cpp
#include"Cube.h"

Cube::Cube() {
	side = 1;
}
Cube::Cube(const double& side) {
	this->side = side;
}
Cube::Cube(const double& side, const string& color, const bool& filled) {
	setSide(side);
	setColor(color);
	setFilled(filled);
}

void Cube::setSide(double newSide) {
	side = newSide;
}
double Cube::getSide() const {
	return side;
}
double Cube::getVolume()  {
	return pow(side, 3);
}
double Cube::getArea()  {
	return 6 * pow(side, 2);
}

string Cube::toString() {
	stringstream ss;
	ss << "cube, " << side;
	return ss.str();
}

File: Python/第15章/15-1/Cube.py
from GeometricObject import GeometricObject

class Cube(GeometricObject):
    def __init__(self, length ,color):
        self.__length = length 
        super().setColor(color)

    def setLength(self, length):
        self.__length = length
    def getLength(self):
        return self.__length

    def getVolume(self):
        return self.__length **3
    def getArea(self):
        return 6* self.__length**2